Cake.Recipe/Content/codecov.cake

```diff
@@ -0,0 +1,33 @@
+///////////////////////////////////////////////////////////////////////////////
+// TASK DEFINITIONS
+///////////////////////////////////////////////////////////////////////////////
+
+BuildParameters.Tasks.UploadCodecovReportTask = Task("Upload-Codecov-Report")
+    .WithCriteria(() => FileExists(BuildParameters.Paths.Files.TestCoverageOutputFilePath))
+    .WithCriteria(() => BuildParameters.IsMainRepository)
+    .WithCriteria(() => BuildParameters.CanPublishToCodecov)
+    .Does(() => RequireTool(CodecovTool, () => {
+        var settings = new CodecovSettings {
+            Files = new[] { BuildParameters.Paths.Files.TestCoverageOutputFilePath.ToString() },
+            Token = BuildParameters.Codecov.RepoToken,
```

Comment:
yup, remember. You tested it in a fork yourself right?
You never added a token.
As long as it's a public repo on Github/Gitlab/Bitbucket (last two I'm a little unsure of) and it's running on appveyor you don't need a token.
Other codecov tools support other CI's as well though.